File: remote/src/main/java/com/kelsos/mbrc/rest/requests/PositionRequest.java
package com.kelsos.mbrc.rest.requests;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;
import org.apache.commons.lang.builder.EqualsBuilder;
import org.apache.commons.lang.builder.HashCodeBuilder;
import org.apache.commons.lang.builder.ToStringBuilder;

@JsonInclude(JsonInclude.Include.NON_NULL)

@JsonPropertyOrder({
    "position"
})
public class PositionRequest {

  @JsonProperty("position") private int position;

  /**
   * @return The position
   */
  @JsonProperty("position") public int getPosition() {
    return position;
  }

  /**
   * @param position The position
   */
  @JsonProperty("position") public PositionRequest setPosition(int position) {
    this.position = position;
    return this;
  }

  @Override public String toString() {
    return ToStringBuilder.reflectionToString(this);
  }

  @Override public int hashCode() {
    return new HashCodeBuilder().append(position).toHashCode();
  }

  @Override public boolean equals(Object other) {
    if (other == this) {
      return true;
    }
    if (!(other instanceof PositionRequest)) {
      return false;
    }
    PositionRequest rhs = ((PositionRequest) other);
    return new EqualsBuilder().append(position, rhs.position).isEquals();
  }
}

File: mbrc-lib/src/main/kotlin/com/kelsos/mbrc/dto/requests/PositionRequest.kt
package com.kelsos.mbrc.dto.requests

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonPropertyOrder

@JsonInclude(JsonInclude.Include.NON_NULL)

@JsonPropertyOrder("position")
class PositionRequest {

    @JsonProperty("position") private var position: Int = 0

    /**
     * @return The position
     */
    @JsonProperty("position") fun getPosition(): Int {
        return position
    }

    /**
     * @param position The position
     */
    @JsonProperty("position") fun setPosition(position: Int): PositionRequest {
        this.position = position
        return this
    }

}
